pyinstrument/low_level/stat_profile.c

```diff
@@ -265,6 +269,30 @@ call_target(ProfilerState *pState, PyFrameObject *frame, int what, PyObject *arg
     Py_DECREF(args);
     return result;
 }
+#else
+static PyObject *
+call_target(ProfilerState *pState, PyFrameObject *frame, int what, PyObject *arg)
+{
+    PyObject *whatstr;
+    PyObject *result;
+
+    whatstr = whatstrings[what];
+    Py_INCREF(whatstr);
+
+    PyFrame_FastToLocals(frame);
+    Py_INCREF(frame);
+
+    PyObject *callargs[4] = { NULL, (PyObject *) frame, whatstr, arg == NULL ? Py_None : arg };
+
+    result = PyObject_Vectorcall(pState->target, callargs + 1, 3 | PY_VECTORCALL_ARGUMENTS_OFFSET, NULL);
```

Comment:
I wonder if [PyObject_CallFunctionObjArgs](https://docs.python.org/3/c-api/call.html#c.PyObject_CallFunctionObjArgs) would be neater here?

Reply:
There's a blurb about the deprecation of the PyEval_CallObject* routines in favor of PyObject_CallObject* in the Python 3.9 release notes. The vectorcall variant is supposed to be the fastest when you can arrange the preconditions right, since it doesn't require construction of a tuple. The extra NULL slot at the beginning of the callargs array allows reused of the structure when using bound methods, I think.

Reply:
Ah, cool, I'll take a look. Btw PyObject_CallFunctionObjArgs is one of the new APIs, I think

Reply:
Oh, I see, the vectorcall calling convention allows for more efficient bound method calling. Cool, let's keep this.